test/index.js

```diff
@@ -17,32 +14,30 @@ var testPages = fs.readdirSync(testPageRoot).map(function(dir) {
 });
 
 describe("Test page", function() {
-  var oldLibPath = process.env.READABILITY_LIB_PATH;
-  process.env.READABILITY_LIB_PATH = path.join(__dirname, "..", "Readability.js");
-  var scrape = require("readable-proxy").scrape;
   testPages.forEach(function(testPage) {
     describe(testPage.dir, function() {
-      it("should render as expected", function() {
-        // Allows up to 10 seconds for parsing to complete.
-        // XXX: Scraping is damn slow. Investigate.
-        this.timeout(10000);
+      it("should render as expected", function(done) {
         var expected = fs.readFileSync(testPage.expected, {encoding: "utf-8"});
-        return scrape("file://" + testPage.source).catch(function(err) {
-          throw err;
-        }).then(function(result) {
-          // print Readability log messages
-          (result.consoleLogs || [])
-            .filter(function(logMessage) {
-              return logMessage.indexOf("Reader: (Readability)") === 0;
-            })
-            .forEach(function(logMessage) {
-              console.log("[LOG]", logMessage);
-            });
-          // normalize html
-          return prettyPrint(result.content);
-        }).should.eventually.become(prettyPrint(expected));
+        jsdom.env(
+          fs.readFileSync(testPage.source, {encoding: "utf-8"}),
+          [path.join(__dirname, "..", "Readability.js")],
+          function (errors, window) {
+            if (errors) {
+              throw new Error(errors);
+            }
+            var uri = {
+              spec: "http://fakehost/test/page.html",
+              host: "fakehost",
+              prePath: "http://fakehost",
+              scheme: "http",
+              pathBase: "http://fakehost/test"
+            };
```

Comment:
Unrelated, but I guess I've always been a bit confused by this `uri` object thing.
It seems that all the fields can be extracted from the `spec` string. It'd be nice if there were just some helper function which converts a fully qualified URL into a `host`, `prePath`, `scheme`, and `pathBase` for you.

I'm guessing this would be pretty trivial w/ Node.js' `url.parse()` function, but it may fit better in the Readability.js library.

```js
var url = require('url');

function uriFromString(str) {
  var urlObj = url.parse(str, true);
  var protocol = urlObj.protocol.replace(':', '');
  var prePath = protocol + '://' + urlObj.hostname;
  var uri = {
    spec: urlObj.href,
    host: urlObj.hostname,
    pathBase: prePath + urlObj.pathname.replace(/[^\/]+$/, ''),
    prePath: prePath,
    scheme: protocol
  };
  return uri;
}

console.log(JSON.stringify(uriFromString(uri.spec), null, 2));
```

I think we've also used http://medialize.github.io/URI.js/ in other projects (via Bower).